### DISCUSSION TITLE
KAMP "not using Firmware retraction"

### DISCUSSION THREAD

**Discussion by ChrSch84** [bassamanator/Sovol-SV06-firmware#153] [General]
Hello all,

i dont know if this is a Problem or not.
With KAMP enable i get this Messages in my Log.

`KAMP purge starting at 135.0, 129.9 and purging 30.0mm of filament, requested flow rate is 12.0mm3/s.
KAMP purge is not using firmware retraction, it is recommended to configure it.`

Any Idea?
Thanks

**Comment by bassamanator** [comment]
I looked into firmware retraction but saw no real benefit of using it, maybe there are benefits?

In any case, you can enable firmware retraction in your slicer, and add the appropriate section in your printer.cfg, see docs [here](https://www.klipper3d.org/Config_Reference.html#firmware_retraction).

**Comment by bassamanator** [comment]
I looked into this, and found this [video](https://youtu.be/T4N59u4wm-Y?si=GcmxCPDZ6djIaMOB) very useful. As someone who likes to recycle g-code, firmware retraction looks awesome, I'll be experimenting with it at some point.

I read the KAMP [code](https://github.com/bassamanator/Klipper-Adaptive-Meshing-Purging/blob/main/Configuration/Line_Purge.cfg) relating to this, and it's not something you need to be concerned about. To put it simply, KAMP would rather not assume what your retraction settings are to provide you with the best, non-drippy nozzle, experience possible, and to do that it needs to know what your retraction settings are, but it cannot know that if you don't have this in your `printer.cfg`. This doesn't affect any SV06/* extruder users because the value is set to our retraction value, 0.5mm.

Best.

**Comment by ChrSch84** [comment]
Thx for the info.

So i can ignore it at the moment :)
